### PR TITLE
Eslint and eslint fixes

### PR DESCRIPTION
**Pull request by kubaGalinski**
Pr with implemented linter and fixed almost all linter problems, there are some rules that i turned off for now like prop-mutating,  multi-word-component-names. I didn't wanted to spend much time on this and refactor lot of components to let those rules pass, i wanted to introduce most necessary rules first. Rest will be done in nearest fututre

### CODE REVIEW

File: src/components/ComponentsSVG/SVG_status_list.vue [hunk 1]
```diff
@@ -1,54 +1,54 @@
...
+    :width="width"
   />
 </template>
 
 <script>
+import reportItemFlags from '../mixins/reportItemFlags.js'
+
 import SVG_building_condition from './SVG_building_condition.vue'
-import SVG_hospital from './SVG_hospital.vue'
+import SVG_car_entrance from './SVG_car_entrance.vue'
```
Comment (sathoune) [inline src/components/ComponentsSVG/SVG_status_list.vue:44]: **Question:** Given that the SVG names were changed, shouldn't we also have the same naming convention here? I'd see that addressed in a separate PR or as we go forward.

File: package.json [hunk 2]
```diff
@@ -2,18 +2,19 @@
   "name": "progect2",
   "private": true,
   "version": "0.0.0",
-  "type": "module",
+  "type": "commonjs",
```
[the remark this answers is not in the document]
Comment (philipkoshy) [reply package.json:5]: Isn't changing the 'type' to commonjs a fairly big change? I thought 'module' was the newer format, i.e., 'ES Modules.'

File: src/components/SidebarComponents/AidWorkerSidebar/MyReportRequestList.vue [hunk 3]
```diff
@@ -3,14 +3,16 @@
...
     </div>
-    <ReportRequestListItem
-      v-if="myUnreviewedMarkers.length > 0"
-      v-for="item in myUnreviewedMarkers"
-      :key="`request${item.id}`"
-      :location-request="item"
-      itemUsageTabName="myRequestsList"
-      @remove-from-my-list="OnRemoveFromMyList"
-    />
+    <div v-if="myUnreviewedMarkers.length > 0">
```
[the remark this answers is not in the document]
Comment (philipkoshy) [reply src/components/SidebarComponents/AidWorkerSidebar/MyReportRequestList.vue:6]: This is ChatGPT 4's response to your inquiry:

You are correct that in some cases, using v-if might not be strictly necessary. If the content inside the <div> element relies on iterating over the unreviewedMarkers array using v-for and the array is empty, nothing would render for that specific iteration.

However, using v-if can still be beneficial in this situation for a few reasons:

Performance: By using v-if, Vue will not even create or process the <div> element and its children if the condition is false. This can lead to performance improvements, particularly when the content inside the <div> is complex or has many children.

Readability: By using v-if, you make the intention of your code clearer to other developers. They can see at a glance that the element is meant to be conditionally rendered based on the length of the array.

Additional content: If the <div> element contains other content or markup that does not rely on iterating over the unreviewedMarkers array, using v-if ensures that this content will also not be rendered when the array is empty.

So, while using v-if might not always be necessary, it can be a good practice to include it for the reasons mentioned above.

[the remark this answers is not in the document]
Comment (sathoune) [reply src/components/SidebarComponents/AidWorkerSidebar/MyReportRequestList.vue:6]: Here, I generated for you a response by the same collective intelligence:

there are a few counterarguments to consider:

    Redundancy: If the only content inside the <div> element is the one using v-for to iterate over the array, then the v-if directive is somewhat redundant, as an empty array would not render any content anyway. In this case, you might decide that the performance and readability benefits are negligible.

    Simplicity: Removing the v-if directive can make the template code simpler and shorter. This might be more appealing in situations where the template is already complex or if you prioritize brevity over explicitness.

    Overhead: Adding the v-if directive introduces a small overhead in terms of processing the condition. In most cases, this overhead is minimal and doesn't impact performance significantly. However, if you are working on an application where every bit of performance optimization matters, you might consider the extra check unnecessary.

    Implicit understanding: Experienced Vue developers may already understand that an empty array in a v-for directive would result in no content being rendered. In this case, the v-if directive could be seen as unnecessary for readability.

In summary, while there are valid reasons to include the v-if directive, there are also counterarguments related to redundancy, simplicity, and overhead. It's essential to consider the specific context of your application, the complexity of the template, and the experience of your development team when deciding whether to include the v-if directive in this scenario.
